crates/rpc-client/src/withdrawal.rs

```diff
@@ -0,0 +1,265 @@
+use anyhow::{bail, Result};
+use ckb_types::prelude::{Entity, Reader};
+use gw_types::bytes::Bytes;
+use gw_types::core::ScriptHashType;
+use gw_types::offchain::CellInfo;
+use gw_types::packed::{
+    Byte32, Script, ScriptReader, WithdrawalLockArgs, WithdrawalLockArgsReader,
+};
+use gw_types::prelude::{Pack, Unpack};
+
+pub fn verify_unlockable_to_owner(
+    info: &CellInfo,
+    last_finalized_block_number: u64,
+    l1_sudt_script_hash: &Byte32,
+) -> Result<()> {
+    verify_l1_sudt_script(info, l1_sudt_script_hash)?;
+    verify_finalized_owner_lock(info, last_finalized_block_number)
+}
+
+fn verify_l1_sudt_script(info: &CellInfo, l1_sudt_script_hash: &Byte32) -> Result<()> {
+    if let Some(sudt_type) = info.output.type_().to_opt() {
+        if info.data.len() < ckb_types::packed::Uint128::TOTAL_SIZE {
+            bail!("invalid l1 sudt data len");
+        }
+
+        if &sudt_type.code_hash() != l1_sudt_script_hash
+            || sudt_type.hash_type() != ScriptHashType::Type.into()
+        {
+            bail!("invalid l1 sudt script");
+        }
+    }
+
+    Ok(())
+}
+
+fn verify_finalized_owner_lock(info: &CellInfo, last_finalized_block_number: u64) -> Result<()> {
+    let args: Bytes = info.output.lock().args().unpack();
```

Comment:
All good, except this.
I think we should keep this implementation uniform. BTW this is a very black box.
How about extracting the same logic into crates/utils/src/withdrawal.rs. And verify there?

Reply:
```gw-utils``` current deps on ```gw-rpc-client``` and ```gw-poa```. We cannot extract this logic into ```gw-utils``` and use it here.
This will result in cyclic package dependency. We may refactor code in the future pr.